### PR TITLE
CMake: Initial CmakeList and CPack template

### PR DESCRIPTION
**Pull request by oliv3r**
This is a preliminary (untested) version of the most basic CMake and
CPack file to create a first release of a package.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
Maybe include some documentation for that this is intended for firmware packages only?

It's not of any significance to Cura, I'd say.

**Comment by oliv3r** [comment]
@Ghostkeeper it's a simple CMake template, for now, it is purely with embedded in mind yes (firmware, okuda, cura-connect) so there's a little bit of 'cura' there :p though I agree that a cura (plugin)  cmake template would be vastly different.

Anyhow, what do you suggest, putting them under a subdir so that a we can have a template for cura as well?

File: templates/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    execute_process(COMMAND ${GIT_EXECUTABLE} describe --dirty --tags --always OUTPUT_VARIABLE GIT_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+endif()
+
+install(
+    FILES "${CMAKE_SOURCE_DIR}/LICENSE"
+    DESTINATION "${CMAKE_INSTALL_FULL_DOCDIR}"
+)
+
+if(EXISTS "${CMAKE_ROOT}/Modules/CPack.cmake")
+    set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Ultimaker dummy template")
```
Comment (oliv3r) [inline templates/CMakeLists.txt]: add git version to summary

**Comment by Ghostkeeper** [comment]
> Anyhow, what do you suggest, putting them under a subdir so that a we can have a template for cura as well?

What I wanted to suggest was to add a bit of documentation in the file or the file name that it's meant for stuff that gets installed on the printer.

**Comment by oliv3r** [comment]
@Ghostkeeper I did not want to over indulge in documentation, as these files tend to get copy pasted without reading. So now it's in the subdirectory embedded and we can always add a README file if more explanation is needed.

**Pull request merged**